test/orange/Orange.test.cc

```diff
@@ -179,4 +179,26 @@ TEST_F(TwoVolumeTest, persistence)
         EXPECT_EQ(SurfaceId{0}, geo.surface_id());
         EXPECT_VEC_SOFT_EQ(Real3({-1.5, 0, 0}), geo.pos());
     }
+    {
+        auto geo = this->make_track_view();
+        EXPECT_EQ(VolumeId{0}, geo.volume_id());
+        EXPECT_EQ(SurfaceId{0}, geo.surface_id());
+        EXPECT_VEC_SOFT_EQ(Real3({-1.5, 0, 0}), geo.pos());
+        geo.move_internal({-1.5, .5, .5});
+    }
+    {
+        auto geo = this->make_track_view();
+        EXPECT_VEC_SOFT_EQ(Real3({-1.5, .5, .5}), geo.pos());
+        geo.set_dir({1, 0, 0});
+    }
+    {
+        auto geo = this->make_track_view();
+        EXPECT_VEC_SOFT_EQ(Real3({1, 0, 0}), geo.dir());
+        EXPECT_SOFT_EQ(3.0, geo.find_next_step());
```

Comment:
Now I'm confused about this one. If the track is at (-1.5, 0.5, 0.5) heading to the right, shouldn't it hit the sphere?

Reply:
Ok, I think I see what's going on. When we call `move_internal`, it doesn't invalidate/change `local_.surface`, so when we go to calculate intersections for the sphere, it still thinks it's on the surface even though it's not.

Reply:
Damn, another good catch: that should reset the surface. I clearly didn't test that function enough 🤦